src/langserver/debugger/evaluate.rs

```diff
@@ -1,45 +1,68 @@
-use super::*;
 use super::dap_types::*;
+use super::*;
 
 const EVALUATE_HELP: &str = "
 #dis, #disassemble: show disassembly for current stack frame";
 
 impl Debugger {
-    pub fn evaluate(&mut self, params: EvaluateArguments) -> Result<EvaluateResponse, Box<dyn std::error::Error>> {
+    pub fn evaluate(
+        &mut self,
+        params: EvaluateArguments,
+    ) -> Result<EvaluateResponse, Box<dyn std::error::Error>> {
         let input = params.expression.trim_start();
         if input.starts_with("#help") {
             return Ok(EvaluateResponse::from(EVALUATE_HELP.trim()));
         }
 
-        let extools = self.extools.get()?;
+        match &mut self.client {
+            DebugClient::Extools(extools) => {
+                let extools = extools.get()?;
 
-        guard!(let Some(frame_id) = params.frameId else {
-            return Err(Box::new(GenericError("Must select a stack frame to evaluate in")));
-        });
+                guard!(let Some(frame_id) = params.frameId else {
+                    return Err(Box::new(GenericError("Must select a stack frame to evaluate in")));
+                });
 
-        let (thread, frame_no) = extools.get_thread_by_frame_id(frame_id)?;
+                let (thread, frame_no) = extools.get_thread_by_frame_id(frame_id)?;
 
-        if input.starts_with('#') {
-            if input == "#dis" || input == "#disassemble" {
-                guard!(let Some(frame) = thread.call_stack.get(frame_no) else {
-                    return Err(Box::new(GenericError("Stack frame out of range")));
-                });
+                if input.starts_with('#') {
+                    if input == "#dis" || input == "#disassemble" {
+                        guard!(let Some(frame) = thread.call_stack.get(frame_no) else {
+                            return Err(Box::new(GenericError("Stack frame out of range")));
+                        });
 
-                let bytecode = extools.bytecode(&frame.proc, frame.override_id);
-                return Ok(EvaluateResponse::from(Self::format_disassembly(bytecode)));
+                        let bytecode = extools.bytecode(&frame.proc, frame.override_id);
+                        return Ok(EvaluateResponse::from(Self::format_disassembly(bytecode)));
+                    }
+                }
             }
+
+            DebugClient::Auxtools(_) => {}
```

Comment:
Just a TODO or is there something making this difficult?

Reply:
There's just no commands to add yet. You could add a diassemble command fairly easily, but I have only designed our disassembler to produce the correct length for each opcode/operands. The output is mostly correct, but that was a secondary objective.

The disassembler produces more correct output than extools, but it's not formatted. There's still some issues with GetVar (I've only noticed them in (init) methods so far.)

Example output would be:
```
Dism for Proc(/world/proc/enable_debugger)
	0-1: DbgFile(spaceman_dmm.dm)
	2-3: DbgLine(32)
	4-6: PushVal("env")
	7-9: PushVal("EXTOOLS_DLL")
	10-15: Call(RuntimeProcField(World, [], GetConfig), 2)
	16-18: SetVar(Local(0))
	19-20: DbgLine(33)
	21-23: GetVar(Local(0))
	24-24: Test
	25-26: Jz(38)
	27-28: DbgLine(34)
	29-31: GetVar(Local(0))
	32-34: PushVal("debug_initialize")
	35-36: CallName(ParamCount(0))
	37-37: Pop
	38-38: End
```

Reply:
The formatting doesn't really matter, but the `#disassemble` command helped me a lot when working on extools.

It's not a priority for this PR, but I wanted to be sure there wasn't anything fundamental blocking it.

The DAP Disassemble request doesn't matter because VSC doesn't implement it.

Reply:
I'll add it in when I get around to figuring out what's wrong with (init) method SetVar instructions (it wasn't GetVar, but they have similar operands so the bug probably exists in both.)